include/deal.II/matrix_free/cuda_hanging_nodes_internal.h

```diff
@@ -758,80 +758,82 @@ namespace CUDAWrappers
-                                                           (fe_degree + 1) +
-                                                         i];
-                        t += w * values[real_idx];
-                      }
+                    const unsigned int real_idx =
+                      (direction == 0) ? index2<fe_degree + 1>(i, y_idx) :
+                                         index2<fe_degree + 1>(x_idx, i);
+
+                    const Number w =
+                      transpose ?
+                        internal::constraint_weights[i * (fe_degree + 1) +
+                                                     interp_idx] :
+                        internal::constraint_weights[interp_idx *
+                                                       (fe_degree + 1) +
+                                                     i];
+                    t += w * values[real_idx];
                   }
-                else
+              }
+            else
+              {
+                for (unsigned int i = 0; i <= fe_degree; ++i)
                   {
-                    for (unsigned int i = 0; i <= fe_degree; ++i)
-                      {
-                        const unsigned int real_idx =
-                          (direction == 0) ? index2<fe_degree + 1>(i, y_idx) :
-                                             index2<fe_degree + 1>(x_idx, i);
-
-                        const Number w =
-                          transpose ?
-                            internal::constraint_weights[(fe_degree - i) *
-                                                           (fe_degree + 1) +
-                                                         fe_degree -
-                                                         interp_idx] :
-                            internal::constraint_weights[(fe_degree -
-                                                          interp_idx) *
-                                                           (fe_degree + 1) +
-                                                         fe_degree - i];
-                        t += w * values[real_idx];
-                      }
+                    const unsigned int real_idx =
+                      (direction == 0) ? index2<fe_degree + 1>(i, y_idx) :
+                                         index2<fe_degree + 1>(x_idx, i);
+
+                    const Number w =
+                      transpose ?
+                        internal::constraint_weights[(fe_degree - i) *
+                                                       (fe_degree + 1) +
+                                                     fe_degree - interp_idx] :
+                        internal::constraint_weights[(fe_degree - interp_idx) *
+                                                       (fe_degree + 1) +
+                                                     fe_degree - i];
+                    t += w * values[real_idx];
                   }
               }
+          }
 
-            __syncthreads();
```

Comment:
... and make sure every thread synchronizes here and the rest is cosmetics?

Reply:
Yeah exactly